src/app/shared/components/template/template-component.ts

```diff
@@ -169,4 +167,25 @@ export class TemplateComponent implements OnInit, AfterContentInit, ITemplateRow
     componentRef.instance.row = row;
     this.componentRef = componentRef;
   }
+
+  /**
+   * If the current template is generated as a child of data_items then updates to
+   * parent variables will not propagate down. Force reprocessing to workaround
+   * See issue https://github.com/IDEMSInternational/open-app-builder/issues/2636
+   */
+  private async hackForceReprocessNestedTemplate() {
+    if (this._row.type === "template") {
+      const componentRef = this.componentRef as ComponentRef<TemplateContainerComponent>;
+      if (componentRef.instance.parent) {
+        // HACK - if parent template changes name of nested child template instance fully
+        // recreate the template container
+        if ((componentRef.instance.templatename(), this._row.value)) {
```

Comment:
This use of the comma operator isn't clear to me. 

I'm supposing that the signal value is being accessed here to ensure that any reactive side-effects (including ultimately updating `this._row.value`) have been triggered before accessing the value, is that right? Or else, to achieve the same result of not accessing the value until any signal-based changes have propagated, I can see how this line could be establishing a reactive dependency between this function and the source signal that wouldn't otherwise be there.

If that's right, might it be clearer to access the signal outside of the if statement and add a comment to explain the dependency being established? I think my confusion is mostly down to unfamiliarity with the comma operator, I can see how this could be an established pattern for making sure the first expression has evaluated and any side-effects propagated before evaluating the first expression, but I wouldn't have expected this to be required when using signals.

Reply:
Whoops, that was meant to be a `!==` operator, although when I change it the debug cases fails to update when parent local variable (as was actually expected, and confusing why it was passing)

So I think for now we'll just have to stick with the heavier condition to always fully recreate when the parent of a child template changes (first part of `componentRef.instance.templateName` being truthy), and try to improve a more finegrained reprocessing when refactoring the templating system

Reply:
Updated in https://github.com/IDEMSInternational/open-app-builder/pull/2647

Reply:
As for the visual-tests failing, I see this happening generally [example failed action](https://github.com/IDEMSInternational/open-app-builder/actions/runs/12436096171/job/34723372445)
```sh
[1220/175851.640539:FATAL:zygote_host_impl_linux.cc(117)] No usable sandbox! Update your kernel or see 
https://chromium.googlesource.com/chromium/src/+/main/docs/linux/suid_sandbox_development.md for more information on developing with the SUID sandbox. If you want to live dangerously and need an immediate workaround, you can try using --no-sandbox.
```
so I think I'll give it a couple days to see if it's an issue resolved within github actions (could be an issue related to internal action runner updates), before looking more closely and trying to update the action/dependencies